### DISCUSSION TITLE
node-ip is not set?

### DISCUSSION THREAD

**Discussion by xhejtman** [rancher/rke2#5140] [Q&A]
It seems, that if `disable-cloud-controller: true` is set, `node-ip: '10.16.58.132,xxx:42d3:0:1:58:132'` is not passed to `kubelet` and has to be set manually. Is this a bug or just something messed in our cluster?

**Comment by manuelbuil** [comment]
If you check `journalctl | grep -i "Running kubelet"`, you don't see the node-ip arg with the correct info?

**Reply by xhejtman** [reply]
``` 
Dec 18 12:25:13 kub-b17.priv.cerit-sc.cz rke2[1853626]: time="2023-12-18T12:25:13+01:00" level=info msg="Running kubelet --address=0.0.0.0 --allowed-unsafe-sysctls=net.ipv4.ip_forward,net.ipv6.conf.all.forwarding --alsologtostderr=false --anonymous-auth=false --authentication-token-webhook=true --authorization-mode=Webhook --cgroup-driver=systemd --client-ca-file=/var/lib/rancher/rke2/agent/client-ca.crt --cluster-dns=10.43.0.10,xxx:42cb:8:3:0:a --cluster-domain=cluster.local --config=/var/lib/rancher/rke2/agent/etc/kubelet.config --container-runtime-endpoint=unix:///run/k3s/containerd/containerd.sock --containerd=/run/k3s/containerd/containerd.sock --cpu-manager-policy=static --eviction-hard=imagefs.available<5%,nodefs.available<5% --eviction-minimum-reclaim=imagefs.available=10%,nodefs.available=10% --fail-swap-on=false --feature-gates=InPlacePodVerticalScaling=true --healthz-bind-address=127.0.0.1 --hostname-override=kub-b17.priv.cerit-sc.cz --kube-api-burst=100 --kube-api-qps=100 --kubeconfig=/var/lib/rancher/rke2/agent/kubelet.kubeconfig --log-file=/var/lib/rancher/rke2/agent/logs/kubelet.log --log-file-max-size=50 --logtostderr=false --max-pods=160 --node-labels=storage=local-ssd,cerit.io/zfs=true,cerit.io/beegfs=true --pod-infra-container-image=index.docker.io/rancher/mirrored-pause:3.6 --pod-manifest-path=/var/lib/rancher/rke2/agent/pod-manifests --protect-kernel-defaults=true --read-only-port=0 --reserved-cpus=64-127 --resolv-conf=/etc/resolv.conf --serialize-image-pulls=false --stderrthreshold=FATAL --system-reserved=memory=8Gi --tls-cert-file=/var/lib/rancher/rke2/agent/serving-kubelet.crt --tls-private-key-file=/var/lib/rancher/rke2/agent/serving-kubelet.key --topology-manager-policy=best-effort --v=5"
```

**Comment by manuelbuil** [comment]
After having a quick look at our code, it is something made on purpose https://github.com/k3s-io/k3s/blob/master/pkg/daemons/agent/agent_linux.go#L126-L129. Note that rke2 is consuming that code. I need to investigate a bit on this

**Comment by xhejtman** [comment]
We are running dual-stack on bare-metal configuration, so I disabled the cloud controller.

**Reply by manuelbuil** [reply]
If you don't disable it, we are providing a minimal and embedded cloud-controller. It should work for your use case, even if it is baremetal

**Reply by xhejtman** [reply]
is there any use for this component in bare-metal case? I understand it does not hurt, but is it for anything?

**Reply by brandond** [reply]
It is there to do exactly what you're complaining about not working if you disable it. It handles setting node IPs in an environment that doesn't have a "real" cloud provider integration to look up IPs from. The kubelet has a very limited ability to set its own IPs, and cannot set external IP at all. It also handles removing the uninitialized taint.  This is why we have a built-in cloud provider. Why are you disabling it?
